File: 351002/Yemyalyanenka/Task 330/publisherApi/Models/Entities/Writer.cs
namespace RestApiTask.Models.Entities
{
    public class Writer : IHasId
    {
        public long Id { get; set; }
        public string Login { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
        public string Firstname { get; set; } = string.Empty;
        public string Lastname { get; set; } = string.Empty;
        public string Role { get; set; } = "CUSTOMER"; // ADMIN или CUSTOMER
    }

}

File: 351001/Tereshchenko/RVTaskFull/WriterService/src/main/java/com/writerservice/models/Writer.java
package com.writerservice.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigInteger;

@Entity
@Table(name = "tbl_writer")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Writer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(name = "login", unique = true)
    @Size(min = 2, max = 64)
    public String login;

    @Column(name = "password")
    @Size(min = 8, max = 128)
    public String password;

    @Column(name = "firstname")
    @Size(min = 2, max = 64)
//    @Pattern(regexp = "[A-Za-z]+")
    public String firstname;

    @Column(name = "lastname")
    @Size(min = 2, max = 64)
//    @Pattern(regexp = "[A-Za-z]+")
    public String lastname;

    @Enumerated(EnumType.STRING)
    @Column(name = "role", nullable = false)
    @Builder.Default
    public WriterRole role = WriterRole.CUSTOMER;

}
